api/python/indigo/ml/clustering.py

```diff
@@ -0,0 +1,19 @@
+from sklearn.cluster import SpectralClustering
+
+
+def clustering(
+    assay_values: list[list], method=SpectralClustering, **kwargs
```

Comment:
Please specify list type, maybe like `list[list[Union[int, float]]]`?

Also `method: Type[ClusterMixing]` seems to be reasonable.

Reply:
Done. 
But ClusterMixing caused some IDE highlights. Still thinking about collecting all valid clustering methods into a list for type checking